lib/client/connection.js

```diff
@@ -714,3 +692,43 @@ Connection.prototype._handleSnapshotFetch = function(error, message) {
   delete this._snapshotRequests[message.id];
   snapshotRequest._handleResponse(error, message);
 };
+
+Connection.prototype._handleLegacyInit = function(message) {
+  // If the minor protocol version has been set, we can ignore this legacy
+  // init message, and continue with the updated handshake protocol.
+  if (message.protocolMinor) return this._initializeHandshake();
```

Comment:
I couldn't think of a good way of adding a test case for legacy clients. I assume that's okay?

Reply:
Let's add a special little test suite that runs using its own `package.json` that has a dependency on the old version of ShareDB.